File: bin/cr_panel_genes.py
#!/usr/bin/env python
from pymongo import MongoClient
import bson
import bson.json_util
import json
import sys

#get all genes in each panel in mongodb, db CR, panels
#bifx@courtagen, jje, jw	11062015
'''$DB = Mongo::MongoReplicaSetClient.new(["172.16.10.55","172.16.20.55","172.16.20.56"], :read => :primary_preferred).db("CR")'''


host = ["172.16.10.55","172.16.20.55","172.16.20.56"]
port = "27017"
db = "CR"
limit = 0
collection = 'panels'

#get collection
mongo = MongoClient(host,int(port))
m_db = mongo[db]
m_collect = m_db[collection]

for panel in m_collect.find().limit(limit):
	#json.dumps(record, sort_keys=True, indent=4, separators=(',', ': '))
	#rec = bson.json_util.dumps(record)
	#output = [panel['_id'], panel[]
	
	if 'genes' in panel:
		for gene in panel['genes']:

			print '\t'.join([str(gene['chr']), str(gene['start']), str(gene['end']), str(gene['hugoname']), str(panel['_id'])])

			#print panel['_id'] + '\t' +  '\t'.join(geneinfo)

#	else:
#		print panel['_id'] + '\tNone\tNone\tNone\tNone\tNone'	

exit()

File: bin/cr_panel_genes.rb
###!/usr/bin/env/ ruby
require 'mongo'
include Mongo

#make bedfile of targets for specific panel
panel_name = "episeektriome_v2"

dbname = "CR"
hosts = ["172.16.10.55:27017", "172.16.20.55:27017", "172.16.20.56:27017"]
collection = "project_types"

$mdb_ptype = Mongo::MongoReplicaSetClient.new(hosts)[dbname][collection]

criteria = {:_id=>panel_name}
attribs = ['targets']

rec = $mdb_ptype.find_one(criteria, fields: attribs)

if rec.nil?
  message = "ERROR: no record with id " + panel_name
  puts message
  exit
end

targets = rec['targets']

targets.each do |target|
	line = "chr" + target["chr"].to_s + "\t" + target["start"].to_s + "\t" + target["end"].to_s + "\t" + target["hugogene"] + "::" + target['_id'] + "\t0\n"
	print line
end

